Gems/ROS2/Code/Source/RobotImporter/URDF/VisualsMaker.cpp

```diff
@@ -171,10 +171,9 @@ namespace ROS2
                 const AZ::Vector3 scaleVector = URDF::TypeConversions::ConvertVector3(meshGeometry->Scale());
 
                 const auto asset = PrefabMakerUtils::GetAssetFromPath(*m_urdfAssetsMapping, AZStd::string(meshGeometry->Uri().c_str()));
-                AZ::Data::AssetId assetId;
+                assetId = Utils::GetModelProductAssetId(asset->m_sourceGuid);
```

Comment:
`asset` is accessed now before/regardless of it is valid (checked by `if (asset)`), not sure if this is intended.

Reply:
Nope, not intended, fixed!